lib/ansible/module_utils/facts/hardware/freebsd.py

```diff
@@ -173,8 +173,41 @@ def get_device_facts(self):
 
         sysdir = '/dev'
         device_facts['devices'] = {}
-        drives = re.compile(r'(ada?\d+|da\d+|a?cd\d+)')  # TODO: rc, disks, err = self.module.run_command("/sbin/sysctl kern.disks")
-        slices = re.compile(r'(ada?\d+s\d+\w*|da\d+s\d+\w*)')
+        # TODO: rc, disks, err = self.module.run_command("/sbin/sysctl kern.disks")
+        drives = re.compile(
+            r"""(
+                ada?\d+|  # ATA/SATA disk device
+                da\d+|    # SCSI disk device
+                a?cd\d+|  # SCSI CDROM drive
+                amrd\d+|  # AMI MegaRAID drive
+                idad\d+|  # Compaq RAID array
+                ipsd\d+|  # IBM ServeRAID RAID array
+                md\d+|    # md(4) disk device
+                mfid\d+|  # LSI MegaRAID SAS array
+                mlxd\d+|  # Mylex RAID disk
+                twed\d+|  # 3ware ATA RAID array
+                vtbd\d+   # VirtIO Block Device
+            )
+            """
+        )
+
+        slice = re.compile(
+            r"""(
+                ada?\d+[ps]\d+\w*|  # ATA/SATA disk device
+                a?cd\d+[ps]\d+\w*|  # SCSI CDROM drive
+                amrd\d+[ps]\d+\w*|  # AMI MegaRAID drive
+                da\d+[ps]\d+\w*|    # SCSI disk device
+                idad\d+[ps]\d+\w*|  # Compaq RAID array
+                ipsd\d+[ps]\d+\w*|  # IBM ServeRAID RAID array
+                md\d+[ps]\d+\w*|    # md(4) disk device
+                mfid\d+[ps]\d+\w*|  # LSI MegaRAID SAS array
+                mlxd\d+[ps]\d+\w*|  # Mylex RAID disk
+                twed\d+[ps]\d+\w*|  # 3ware ATA RAID array
+                vtbd\d+[ps]\d+\w*   # VirtIO Block Device
```

Comment:
Looks like `re.VERBOSE` flag (or it's `(?x)` variant) is missing. Without it, multi-lining wouldn't work. Perhaps, a test would be useful to catch such things..

Additionally, the variable got renamed accidentally (with the trailing `s` being dropped) and the control flow errors out elsewhere.

Finally, I noticed that there's a lot of duplication that can hopefully be addressed as well, with the following patch:

```suggestion
            r"""(?x)(
              (?:
                ada?   # ATA/SATA disk device
                |da    # SCSI disk device
                |a?cd  # SCSI CDROM drive
                |amrd  # AMI MegaRAID drive
                |idad  # Compaq RAID array
                |ipsd  # IBM ServeRAID RAID array
                |md    # md(4) disk device
                |mfid  # LSI MegaRAID SAS array
                |mlxd  # Mylex RAID disk
                |twed  # 3ware ATA RAID array
                |vtbd  # VirtIO Block Device
              )\d+
            )
            """
        )

        slice = re.compile(
            r"""(?x)(
              (?:
                ada?   # ATA/SATA disk device
                |a?cd  # SCSI CDROM drive
                |amrd  # AMI MegaRAID drive
                |da    # SCSI disk device
                |idad  # Compaq RAID array
                |ipsd  # IBM ServeRAID RAID array
                |md    # md(4) disk device
                |mfid  # LSI MegaRAID SAS array
                |mlxd  # Mylex RAID disk
                |twed  # 3ware ATA RAID array
                |vtbd  # VirtIO Block Device
              )\d+[ps]\d+\w*
```

*(I haven't tested this so there may be typos)*